### PR TITLE
Add a setup.cfg parser

### PR DESCRIPTION
**Pull request by dmerejkowsky**
Not used for much, but could be later used to:

* Properly build venv paths when using DMENV_VENV_OUTSIDE_PROJECT
* Implement things like `dmenv add`, `dmenv remove` by round-tripping contents of the `setup.cfg` file
* And a few other good stuff

TODO:

* [x] Check the parser by running on real-world setup.cfg files
* [x] Better error on sections looking like `[this` or `that]` 
* [x] Line number in error message

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,9 @@ version = "0.18.0"
 app_dirs = "1.2.1"
 colored = "1.6"
 libc = "0.2.46"
+serde = "*"
```
Comment (tux3) [inline Cargo.toml]: You probably want to write 1.0 ?

File: Cargo.toml [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
+app_dirs = "1.2.1"
 colored = "1.6"
-structopt = "0.2"
-which = "2.0"
 libc = "0.2.46"
 regex = "1.1.0"
-app_dirs = "1.2.1"
-smallvec = "0.6.10"
 rust-ini = "0.13.0"
+smallvec = "0.6.10"
```
Comment (tux3) [inline Cargo.toml]: I don't think you use smallvec directly, you can remove it from the dependencies.

File: Cargo.toml [hunk 3]
```diff
@@ -15,6 +15,9 @@ version = "0.18.0"
 app_dirs = "1.2.1"
 colored = "1.6"
 libc = "0.2.46"
+serde = "*"
+serde_derive = "*"
```
Comment (tux3) [inline Cargo.toml]: And I don't think you use `serde_derive` actually?

**Comment by dmerejkowsky** [comment]
Well, this implementation sucks:

* Too many copies
* Too much hand-written code
* No feature-parity with setup.cfg
* Will probably _not_ allow round-trip implementation

Let's close this and try again later

**Pull request closed without merging**